appstore/validator_test.go

```diff
@@ -183,7 +183,7 @@ func TestVerifyBadURL(t *testing.T) {
 }
 
 func TestVerifyBadPayload(t *testing.T) {
-	s := httptest.NewServer(serverWithResponse(`{"status": 21002}`))
+	s := httptest.NewServer(serverWithResponse(http.StatusBadRequest, `{"status": 21002}`))
```

Comment:
when status=21002 is returned from the server will it be a BadRequest? I assume that would still be ok(200).
I think what we should do is to add more tests to test status code alone.(ex: TestVerifyBadRequest, TestVerifyInternalServerError ...)

Reply:
I am also wondering, which test case is testing the success case?
Success means, ```err != nil && resp.Status == 0``` right?

Reply:
I can add more tests, but will it matter? The code itself doesn't seem to care about the status code, but rather the body.

Reply:
Would you mind telling me which test case is testing the success case?

Reply:
And, yes, it matters. As a testcase, this mock is misleading as there is never a case BadRequest and {status: 21002} can be returned in combination.

Reply:
Success case added. Misread your original comment.. my bad. I fixed that test case. 

Refactored the tests (not comments as the commit suggested - mixed it up with another ticket that I was working on!) to make adding extra test cases easy :)